Python/pylifecycle.c

```diff
@@ -1257,6 +1257,20 @@ Py_FinalizeEx(void)
     PyThreadState *tstate = _PyRuntimeState_GetThreadState(runtime);
     PyInterpreterState *interp = tstate->interp;
 
+    // Finalize sub-interpreters.
+    runtime->interpreters.finalizing = 1;
+    PyInterpreterState *subinterp = PyInterpreterState_Head();
+    PyInterpreterState *next_interp;
+    while (subinterp != NULL) {
+        next_interp = PyInterpreterState_Next(subinterp);
+        if (subinterp != PyInterpreterState_Main()) {
```

Comment:
Is what you describe here distinct from bpo-37776? If so I may be missing a subtlety :)

Reply:
That issue is about a test that checks the behavior.  I was talking about actually fixes to pylifecycle.c to enforce the behavior.  Of course, if the tests described in bpo-37776 fail then things would have to be fixed. :smile:  So that probably *is* the right place to also track fixing the problem.  I guess I'd just like a more deliberate description there of what needs to be tested (and enforced).

Reply:
[bpo-38865](https://bugs.python.org/issue38865) seems to track this, which somehow I think I missed back in January!